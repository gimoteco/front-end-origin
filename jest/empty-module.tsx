import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
