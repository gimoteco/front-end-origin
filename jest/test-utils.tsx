import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
