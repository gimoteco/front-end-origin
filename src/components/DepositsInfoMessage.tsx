import * as React from 'react';
import { formatMoney } from '../utils/format';

interface DepositsInfoMessageProps {
  numberOfDeposits: number;
  total: number;
  month: string;
  year: string;
}

function DepositsInfoMessage({
  numberOfDeposits,
  total,
  month,
  year
}: DepositsInfoMessageProps) {
  return (
    <p>
      Youâ€™re planning <strong>{numberOfDeposits} monthly deposits</strong> to
      reach your <strong>{formatMoney(total)}</strong> goal by{' '}
      <strong>
        {month} {year}.
      </strong>
    </p>
  );
}

export default React.memo(DepositsInfoMessage);
