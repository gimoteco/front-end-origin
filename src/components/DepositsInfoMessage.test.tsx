import React from 'react';
import { render } from '../../jest/test-utils';
import DepositsInfoMessage from './DepositsInfoMessage';

describe('DepositsInfoMessage', () => {
  it('should render according snapshot', () => {
    const { container } = render(
      <DepositsInfoMessage
        numberOfDeposits={2}
        total={1000}
        month={'March'}
        year={'1991'}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
