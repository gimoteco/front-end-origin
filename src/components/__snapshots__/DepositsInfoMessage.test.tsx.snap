// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositsInfoMessage should render according snapshot 1`] = `
<div>
  <p>
    Youâ€™re planning 
    <strong>
      2
       monthly deposits
    </strong>
     to reach your 
    <strong>
      $1,000
    </strong>
     goal by
     
    <strong>
      March
       
      1991
      .
    </strong>
  </p>
</div>
`;
